warehouse/upload.go

```diff
@@ -1459,65 +1481,58 @@ func (job *UploadJobT) setUploadError(statusError error, state string) (string,
 		return "", fmt.Errorf("unable to change upload columns: %w", err)
 	}
 
-	inputCount := job.getTotalRowsInStagingFiles()
-	outputCount, _ := job.getTotalEventsUploaded(false)
-	failCount := inputCount - outputCount
-	reportingStatus := jobsdb.Failed.State
-	if state == Aborted {
-		reportingStatus = jobsdb.Aborted.State
-	}
-
-	reportingMetrics := []*types.PUReportedMetric{{
-		ConnectionDetails: types.ConnectionDetails{
-			SourceID:        job.upload.SourceID,
-			DestinationID:   job.upload.DestinationID,
-			SourceBatchID:   job.upload.SourceBatchID,
-			SourceTaskID:    job.upload.SourceTaskID,
-			SourceTaskRunID: job.upload.SourceTaskRunID,
-			SourceJobID:     job.upload.SourceJobID,
-			SourceJobRunID:  job.upload.SourceJobRunID,
-		},
-		PUDetails: types.PUDetails{
-			InPU:       types.BATCH_ROUTER,
-			PU:         types.WAREHOUSE,
-			TerminalPU: true,
-		},
-		StatusDetail: &types.StatusDetail{
-			Status:         reportingStatus,
-			StatusCode:     400, // TODO: Change this to error specific code
-			Count:          failCount,
-			SampleEvent:    []byte("{}"),
-			SampleResponse: string(serializedErr),
-		},
-	}}
-	if outputCount > 0 {
-		reportingMetrics = append(reportingMetrics, &types.PUReportedMetric{
+	job.resetTableUploadStatuses()
+	_, currentJobSucceededTables := job.getTablesToSkip()
+	_, loadFilesTableEventCountMap, err := job.getLoadFilesTablePresenceAndEventCountMaps()
+	if err != nil {
+		loadFilesTableEventCountMap = make(map[tableNameT]int)
+	}
+
+	for tableName := range job.upload.UploadSchema {
+		_, ok := currentJobSucceededTables[string(tableName)]
```

Comment:
UploadSchema is basically map[string]map[string]string. Let's not wrap it again with string. Not required.